React InstantSearch/routing-seo-friendly/src/App.js

```diff
@@ -18,118 +21,171 @@ const searchClient = algoliasearch(
   '6be0576ff61c053d5f9a3225e2a90f76'
 );
 
-const createURL = state => {
-  const brands =
-    state.refinementList &&
-    state.refinementList.brand &&
-    state.refinementList.brand.join('~');
+const encodedCategories = {
+  Cameras: 'Cameras & Camcorders',
+  Cars: 'Car Electronics & GPS',
+  Phones: 'Cell Phones',
+  TV: 'TV & Home Theater',
+};
 
+const decodedCategories = Object.keys(encodedCategories).reduce((acc, key) => {
+  const newKey = encodedCategories[key];
+  const newValue = key;
+
+  return {
+    ...acc,
+    [newKey]: newValue,
+  };
+}, {});
+
+// Returns a slug from the category name.
+// Spaces are replaced by "+" to make
+// the URL easier to read and other
+// characters are encoded.
+function getCategorySlug(name) {
+  const encodedName = decodedCategories[name] || name;
+
+  return encodedName
+    .split(' ')
+    .map(encodeURIComponent)
+    .join('+');
+}
+
+// Returns a name from the category slug.
+// The "+" are replaced by spaces and other
+// characters are decoded.
+function getCategoryName(slug) {
+  const decodedSlug = encodedCategories[slug] || slug;
+
+  return decodedSlug
+    .split('+')
+    .map(decodeURIComponent)
+    .join(' ');
+}
+
+const createURL = state => {
   const isDefaultRoute =
     !state.query &&
     state.page === 1 &&
     state.refinementList.brand &&
```

Comment:
`state.refinementList` is always defined?

Reply:
Yes, actually we can even simplify the condition to:

```js
const isDefaultRoute =
  !state.query &&
  state.page === 1 &&
  state.refinementList.brand.length === 0 &&
  !state.menu.categories;
```

Reply:
It's always defined in our case because the `RefinementList` is always mounted. Since it's an example I would over check because end users might have UIs that does not always have those widgets.